lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -790,7 +791,13 @@ ParseResult FIRParser::parseEnumType(FIRRTLType &result) {
                  "expected leading '{|' in enumeration type"))
     return failure();
   SmallVector<FEnumType::EnumElement> elements;
+  bool first = true;
   if (parseListUntil(FIRToken::r_brace_bar, [&]() -> ParseResult {
+        if (!first) {
+          if (parseToken(FIRToken::comma, "expected ','")) {
+            return failure();
+          }
+        }
```

Comment:
I think every parseListUntil use is modified in the PR, can we change `parseListUntil` to something like `parseCommaSeparatedListUntil` and move `first` stuff to that function?

Reply:
Good catch. I didn't realize this was locally-defined.

I plan on bringing the parser into alignment with the grammar in the spec, so I expect this to change. As long as this looks correct, I will probably leave it as it is for now.